providers/src/airflow/providers/sftp/hooks/sftp.py

```diff
@@ -275,6 +276,47 @@ def delete_file(self, path: str) -> None:
         conn = self.get_conn()
         conn.remove(path)
 
+    def retrieve_directory(self, remote_full_path: str, local_full_path: str, prefetch: bool = True) -> None:
+        """
+        Transfer the remote directory to a local location.
+
+        If local_full_path is a string path, the directory will be put
+        at that location.
+
+        :param remote_full_path: full path to the remote directory
+        :param local_full_path: full path to the local directory
+        :param prefetch: controls whether prefetch is performed (default: True)
+        """
+        os.mkdir(local_full_path)
```

Comment:
@Dawnpool Can we add a check if `local_full_path` exists before trying to create it again?

Reply:
@kunaljubce 
Added it by raising an error if the path already exists. Or, how about just deleting the whole directory and recreating it if it already exists?

Reply:
@kunaljubce I've also updated this check from `isdir()` to `exists()` to make sure both a file and a directory with the same name do not exist.